File: common/api/src/main/java/tech/artcoded/atriangle/api/dto/SparqlQueryRequest.java
package tech.artcoded.atriangle.api.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class SparqlQueryRequest {
  public enum SparqlQueryRequestType {
    ASK_QUERY, SELECT_QUERY, CONSTRUCT_QUERY
  }

  private String projectId;
  private String freemarkerTemplateFileId;
  private Map<String, String> variables;
  private SparqlQueryRequestType type;
}

File: frontend/atriangle-frontend/src/app/core/models/sparql.query.request.ts
export interface SparqlQueryRequest {
  projectId: string;
  variables?: Map<string, string>;
  freemarkerTemplateFileId: string;
  type: SparqlQueryRequestType;
}

export enum SparqlQueryRequestType {
  ASK_QUERY, SELECT_QUERY, CONSTRUCT_QUERY
}
